app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java

```diff
@@ -276,4 +286,13 @@ private void fetchHomeDetails() {
             new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS)));
     dialog.setCancelable(false).show();
   }
+
+  @Override public boolean onQueryTextSubmit(String query) {
+    presenter.fetchNewLocation(query);
```

Comment:
from the view's perspective, it is performing a search. Only the presenter should know that the search should be for a location. A appropriate name could be presenter.searchLocation(String query).

Reply:
:ok: